# **BANCO DE DADOS DE INFORMATIVO DO SUPREMO TRIBUNAL FEDERAL** 

O Supremo Tribunal Federal em seu site (https://portal.stf.jus.br/textos/verTexto.asp?servico=informativoSTF) publica os informativos do STF, que consiste em um periódico semanal da Corte Suprema do país que apresenta, de forma objetiva e concisa, resumos das teses e conclusões dos principais julgamentos realizados pelos órgãos colegiados - Plenário e Turmas.

A seleção dos processos noticiados no referido Informativo leva em consideração critérios de relevância, novidade e contemporaneidade da temática objeto de julgamento.

O banco de dados de todos os informativos do STF já publicados encontra-se disponível no seguinte link:

https://www.stf.jus.br/arquivo/cms/informativoSTF/anexo/Informativo_Dados/Dados_InformativosSTF.xlsx

As edições semanais dos Informativos do STF são disponibilizadas em pdf, doc e html, contudo somente através do banco de dados (planilha em excel = Dataset) disponibilizados pelo Supremo Tribunal Federal é possível acessar de única vez várias edições e seus conteúdos.

A proposta do presente projeto é fazer uma análise exploratória deste banco de dados e por meio do Streamlit apresentar um aplicativo de filtragem e de busca de determinado julgamento, de modo a acessar o conteúdo do julgado. 

In [3]:
import pandas as pd
import streamlit as st


In [8]:
InformativosSTF= pd.read_excel('Dados_InformativosSTF.xlsx')
InformativosSTF.shape

(10977, 24)

> **Note-se que nosso Dataset possui 10977 linhas e 24 colunas**.

Nosso Dataset possui os seguintes dados (feactures):



| Colunas | Descrição|
|---------|----------|
|Informativo | indica o número da edição do Informativo do STF |
|Classe Processo | tipo de ação ou recurso |
|Número Processo | indica o número do processo |
|Incidente Julgamento | recurso incidental ou ação incidental |
| UF | Estado originário da ação |
| Observação | Observações acerca do julgamento, por exemplo se o julgamento foi em conjunto com outras ações|
|Data Julgamento | Data em que ocorreu o julgamento |
|Relator | Ministro do Supremo Tribunal Federal que atuou como relator da ação |
|Redator Acórdão| Ministro do Supremo Tribunal Federal que redigiu o acórdão vencedor |
|Órgão Julgador | Plenário ou Turma Julgadora |
|Tipo Julgamento | Virtual ou Presencial|
|Situação Julgamento | Concluído ou Suspenso|
|Título | Título do Resumo |
|Tese Julgado| Tese Jurídica Oficial |
|Resumo | Resumo do Julgado |
|Notícia | Notícia do julgado com ênfase nas conclusões e nos principais fundamentos |
|Ramo do Direito | Ramo do Direito que pode ser mais de um no mesmo julado (direito penal, direito constitucional, direito administrativo, direito tributário, etc.)|
| Matéria | Assunto sobre qual o julgado aborda (por exemplo: isenção fiscal, aposentadoria, licitação, Zona Franca de Manaus, etc.)|
|Repercussão Geral | Se o julgado tem ou não repercussão geral, ou seja, se aplica a todos os julgados semelhantes ou não. A existência de valores nulos ('na') se justifica pelo fato do julgado ser antes da previsão legal de repercussão legal|
|Tema RG | número do tema de Repercussão Geral |
|Legislação | dispositivos legais que o julgado aborda |
|ODS ONU 2030 | classificação de acordo com a ODS ONU 2030 |
| Covid-19 | se o julgado no período da COVID-19 (sim, não, 'na')|
|Notícia Completa | Notícia Completa do Julgado |


In [10]:
InformativosSTF.isna().sum()

Informativo                 0
Classe Processo             0
Número Processo             0
Incidente Julgamento     8798
UF                        143
Observação               9758
Data Julgamento             0
Relator                     0
Redator Acórdão          9614
Órgão Julgador              0
Tipo Julgamento             0
Situação Julgamento         0
Título                      3
Tese Julgado            10289
Resumo                   8462
Notícia                   190
Ramo Direito                1
Matéria                     0
Repercussão Geral           2
Tema RG                 10360
Legislação               3458
ODS ONU 2030            10249
Covid-19                 4926
Notícia completa            1
dtype: int64

In [28]:
InformativosSTF=InformativosSTF.rename(columns = {'Classe Processo': 'Classe_Processo', 'Número Processo': 'Número_Processo', 
                                'Incidente Julgamento':'Incidente_Julgamento','Data Julgamento':'Data_Julgamento',
                                'Redator Acórdão':'Redator_Acórdão', 'Tipo Julgamento':'Tipo_Julgamento', 
                                'Situação Julgamento':'Situação_Julgamento', 'Tese Julgado':'Tese_Julgado',
                                'Ramo Direito':'Ramo_Direito', 'Repercussão Geral':'Repercussão_Geral',
                                'Tema RG':'Tema_RG', 'ODS ONU 2030':'ODS_ONU_2030', 'Notícia completa':'Notícia_Completa'
                                                 })
InformativosSTF['Incidente_Julgamento']=InformativosSTF['Incidente_Julgamento'].fillna(str('nan'))
InformativosSTF['UF']=InformativosSTF['UF'].fillna(str('nan'))
InformativosSTF['Observação']=InformativosSTF['Observação'].fillna(str('nan'))
InformativosSTF['Redator_Acórdão']=InformativosSTF['Redator_Acórdão'].fillna(str('nan'))
InformativosSTF['Título']=InformativosSTF['Título'].fillna(str('nan'))
InformativosSTF['Tese_Julgado']=InformativosSTF['Tese_Julgado'].fillna(str('nan'))
InformativosSTF['Resumo']=InformativosSTF['Resumo'].fillna(str('nan'))
InformativosSTF['Notícia']=InformativosSTF['Notícia'].fillna(str('nan'))
InformativosSTF['Ramo_Direito']=InformativosSTF['Ramo_Direito'].fillna(str('nan'))
InformativosSTF['Repercussão_Geral']=InformativosSTF['Repercussão_Geral'].fillna(str('nan'))
InformativosSTF['Tema_RG']=InformativosSTF['Tema_RG'].fillna(int(0))
InformativosSTF['Tema_RG']=InformativosSTF['Tema_RG'].astype(int)
InformativosSTF['Legislação']=InformativosSTF['Legislação'].fillna(str('nan'))
InformativosSTF['ODS_ONU_2030']=InformativosSTF['ODS_ONU_2030'].fillna(str('nan'))
InformativosSTF['Covid-19']=InformativosSTF['Covid-19'].fillna(str('nan'))
InformativosSTF['Legislação']=InformativosSTF['Legislação'].fillna(str('nan'))
InformativosSTF['Notícia_Completa']=InformativosSTF['Notícia_Completa'].fillna(str('nan'))

> A existência de valores nulos ('na') não afeta o contúdo da notícia, pois demonstra somente que quando do julgamento não havia previsão legal para constar determinado dado, por exemplos, julgados ocorridos antes da previsão legal que possibilidade atribuir repersussão geral ao julgamento ou julgados ocorridos antes da ocorrência da pandemia do Covid-19.

In [29]:
InformativosSTF.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10977 entries, 0 to 10976
Data columns (total 24 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   Informativo           10977 non-null  int64         
 1   Classe_Processo       10977 non-null  object        
 2   Número_Processo       10977 non-null  int64         
 3   Incidente_Julgamento  10977 non-null  object        
 4   UF                    10977 non-null  object        
 5   Observação            10977 non-null  object        
 6   Data_Julgamento       10977 non-null  datetime64[ns]
 7   Relator               10977 non-null  object        
 8   Redator_Acórdão       10977 non-null  object        
 9   Órgão Julgador        10977 non-null  object        
 10  Tipo_Julgamento       10977 non-null  object        
 11  Situação_Julgamento   10977 non-null  object        
 12  Título                10977 non-null  object        
 13  Tese_Julgado    

In [30]:
df= InformativosSTF.copy()
df['Data_Julgamento'] =df['Data_Julgamento'].astype(str)
df[['Ano', 'Mês', 'Dia-hora']]=df['Data_Julgamento'].str.split('-', expand=True)
df[['Dia', 'Hora']] = df['Dia-hora'].str.split(' ', expand=True)
df = df.drop(['Dia-hora','Hora'], axis=1)
df.head(2)

Informativo Classe_Processo  Número_Processo Incidente_Julgamento  UF  \
0         1127             ADI             7218                  nan  PB   
1         1127             ADI             7239                  nan  DF   

  Observação      Data_Julgamento                    Relator Redator_Acórdão  \
0        nan  2024-03-08 03:00:00          MIN. DIAS TOFFOLI             nan   
1        nan  2024-03-08 03:00:00  MIN. LUÍS ROBERTO BARROSO             nan   

  Órgão Julgador  ...                                            Matéria  \
0       Plenário  ...  Organização Político-Administrativa; Administr...   
1       Plenário  ...  Zona Franca de Manaus; isenção fiscal; áreas d...   

  Repercussão_Geral Tema_RG  \
0               Não       0   
1               Não       0   

                                          Legislação  \
0  CF/1988: arts. 132, caput e 207.\nADCT: art. 6...   
1  Decreto-Lei nº 288/1967: art. 37\nLei nº 14.18...   

                                 ODS_ONU_2030 Covid-19  \
0  8 Trabalho Decente e Crescimento Econômico      Não   
1                                         nan      Não   

                                    Notícia_Completa   Ano Mês  Dia  
0  São inconstitucionais — por ofensa ao princípi...  2024  03   08  
1  “É constitucional o dispositivo de lei federal...  2024  03   08  

[2 rows x 27 columns]

In [33]:
df['Ramo_Direito']


0                               Direito Administrativo
1                                   Direito Tributário
2                               Direito Administrativo
3        Direito Administrativo;Direito Constitucional
4                                   Direito Tributário
                             ...                      
10972                           Direito Administrativo
10973                         Direito Processual Penal
10974                                    Direito Penal
10975                           Direito Administrativo
10976    Direito Constitucional;Direito Administrativo
Name: Ramo_Direito, Length: 10977, dtype: object

In [34]:
df1 = df['Ramo_Direito'].astype(str)
df2=df1.str.split(";", expand=True)
# print(df2[0].value_counts())

df2.loc[df2[0]=='Direito Constitucional', 'Direito_Constitucional']=1  
df2.loc[df2[1]=='Direito Constitucional', 'Direito_Constitucional']=1
df2.loc[df2[2]=='Direito Constitucional', 'Direito_Constitucional']=1
df2.loc[df2[3]=='Direito Constitucional', 'Direito_Constitucional']=1
df2.loc[df2[4]=='Direito Constitucional', 'Direito_Constitucional']=1

df2.loc[df2[0]=='Direito Processual Penal', 'Direito_Processual_Penal']=1  
df2.loc[df2[1]=='Direito Processual Penal', 'Direito_Processual_Penal']=1
df2.loc[df2[2]=='Direito Processual Penal', 'Direito_Processual_Penal']=1  
df2.loc[df2[3]=='Direito Processual Penal', 'Direito_Processual_Penal']=1  
df2.loc[df2[4]=='Direito Processual Penal', 'Direito_Processual_Penal']=1  

df2.loc[df2[0]=='Direito Administrativo', 'Direito_Administrativo']=1
df2.loc[df2[1]=='Direito Administrativo', 'Direito_Administrativo']=1 
df2.loc[df2[2]=='Direito Administrativo', 'Direito_Administrativo']=1 
df2.loc[df2[3]=='Direito Administrativo', 'Direito_Administrativo']=1 
df2.loc[df2[4]=='Direito Administrativo', 'Direito_Administrativo']=1

df2.loc[df2[0]=='Direito Penal', 'Direito_Penal']=1
df2.loc[df2[1]=='Direito Penal', 'Direito_Penal']=1
df2.loc[df2[2]=='Direito Penal', 'Direito_Penal']=1
df2.loc[df2[3]=='Direito Penal', 'Direito_Penal']=1
df2.loc[df2[4]=='Direito Penal', 'Direito_Penal']=1

df2.loc[df2[0]=='Direito Processual Civil', 'Direito_Processual_Civil']=1
df2.loc[df2[1]=='Direito Processual Civil', 'Direito_Processual_Civil']=1
df2.loc[df2[2]=='Direito Processual Civil', 'Direito_Processual_Civil']=1
df2.loc[df2[3]=='Direito Processual Civil', 'Direito_Processual_Civil']=1
df2.loc[df2[4]=='Direito Processual Civil', 'Direito_Processual_Civil']=1

df2.loc[df2[0]=='Direito Tributário', 'Direito_Tributário']=1
df2.loc[df2[1]=='Direito Tributário', 'Direito_Tributário']=1
df2.loc[df2[2]=='Direito Tributário', 'Direito_Tributário']=1
df2.loc[df2[3]=='Direito Tributário', 'Direito_Tributário']=1
df2.loc[df2[4]=='Direito Tributário', 'Direito_Tributário']=1

df2.loc[df2[0]=='Direito Previdenciário', 'Direito_Previdenciário']=1
df2.loc[df2[1]=='Direito Previdenciário', 'Direito_Previdenciário']=1
df2.loc[df2[2]=='Direito Previdenciário', 'Direito_Previdenciário']=1
df2.loc[df2[3]=='Direito Previdenciário', 'Direito_Previdenciário']=1
df2.loc[df2[4]=='Direito Previdenciário', 'Direito_Previdenciário']=1

df2.loc[df2[0]=='Direito Eleitoral', 'Direito_Eleitoral']=1
df2.loc[df2[1]=='Direito Eleitoral', 'Direito_Eleitoral']=1
df2.loc[df2[2]=='Direito Eleitoral', 'Direito_Eleitoral']=1
df2.loc[df2[3]=='Direito Eleitoral', 'Direito_Eleitoral']=1
df2.loc[df2[4]=='Direito Eleitoral', 'Direito_Eleitoral']=1

df2.loc[df2[0]=='Direito Internacional', 'Direito_Internacional']=1
df2.loc[df2[1]=='Direito Internacional', 'Direito_Internacional']=1
df2.loc[df2[2]=='Direito Internacional', 'Direito_Internacional']=1
df2.loc[df2[3]=='Direito Internacional', 'Direito_Internacional']=1
df2.loc[df2[4]=='Direito Internacional', 'Direito_Internacional']=1

df2.loc[df2[0]=='Direito do Trabalho', 'Direito_Trabalho']=1
df2.loc[df2[1]=='Direito do Trabalho', 'Direito_Trabalho']=1
df2.loc[df2[2]=='Direito do Trabalho', 'Direito_Trabalho']=1
df2.loc[df2[3]=='Direito do Trabalho', 'Direito_Trabalho']=1
df2.loc[df2[4]=='Direito do Trabalho', 'Direito_Trabalho']=1

df2.loc[df2[0]=='Direito Penal Militar', 'Direito_Penal_Militar']=1
df2.loc[df2[1]=='Direito Penal Militar', 'Direito_Penal_Militar']=1
df2.loc[df2[2]=='Direito Penal Militar', 'Direito_Penal_Militar']=1
df2.loc[df2[3]=='Direito Penal Militar', 'Direito_Penal_Militar']=1
df2.loc[df2[4]=='Direito Penal Militar', 'Direito_Penal_Militar']=1

df2.loc[df2[0]=='Direito Processual Penal Militar', 'Direito_Processual_Penal_Militar']=1
df2.loc[df2[1]=='Direito Processual Penal Militar', 'Direito_Processual_Penal_Militar']=1
df2.loc[df2[2]=='Direito Processual Penal Militar', 'Direito_Processual_Penal_Militar']=1
df2.loc[df2[3]=='Direito Processual Penal Militar', 'Direito_Processual_Penal_Militar']=1
df2.loc[df2[4]=='Direito Processual Penal Militar', 'Direito_Processual_Penal_Militar']=1

df2.loc[df2[0]=='Direito Civil', 'Direito_Civil']=1
df2.loc[df2[1]=='Direito Civil', 'Direito_Civil']=1
df2.loc[df2[2]=='Direito Civil', 'Direito_Civil']=1
df2.loc[df2[3]=='Direito Civil', 'Direito_Civil']=1
df2.loc[df2[4]=='Direito Civil', 'Direito_Civil']=1

df2.loc[df2[0]=='Direito Financeiro', 'Direito_Financeiro']=1
df2.loc[df2[1]=='Direito Financeiro', 'Direito_Financeiro']=1
df2.loc[df2[2]=='Direito Financeiro', 'Direito_Financeiro']=1
df2.loc[df2[3]=='Direito Financeiro', 'Direito_Financeiro']=1
df2.loc[df2[4]=='Direito Financeiro', 'Direito_Financeiro']=1

df2.loc[df2[0]=='Direito da Criança e do Adolescente', 'Direito_Criança_Adolescente']=1
df2.loc[df2[1]=='Direito da Criança e do Adolescente', 'Direito_Criança_Adolescente']=1
df2.loc[df2[2]=='Direito da Criança e do Adolescente', 'Direito_Criança_Adolescente']=1
df2.loc[df2[3]=='Direito da Criança e do Adolescente', 'Direito_Criança_Adolescente']=1
df2.loc[df2[4]=='Direito da Criança e do Adolescente', 'Direito_Criança_Adolescente']=1

df2.loc[df2[0]=='Direito Ambiental', 'Direito_Ambiental']=1
df2.loc[df2[1]=='Direito Ambiental', 'Direito_Ambiental']=1
df2.loc[df2[2]=='Direito Ambiental', 'Direito_Ambiental']=1
df2.loc[df2[3]=='Direito Ambiental', 'Direito_Ambiental']=1
df2.loc[df2[4]=='Direito Ambiental', 'Direito_Ambiental']=1

df2.loc[df2[0]=='Direito Processual do Trabalho', 'Direito_Processual_Trabalho']=1
df2.loc[df2[1]=='Direito Processual do Trabalho', 'Direito_Processual_Trabalho']=1
df2.loc[df2[2]=='Direito Processual do Trabalho', 'Direito_Processual_Trabalho']=1
df2.loc[df2[3]=='Direito Processual do Trabalho', 'Direito_Processual_Trabalho']=1
df2.loc[df2[4]=='Direito Processual do Trabalho', 'Direito_Processual_Trabalho']=1

df2.loc[df2[0]=='Direito da Saúde', 'Direito_Saúde']=1
df2.loc[df2[1]=='Direito da Saúde', 'Direito_Saúde']=1
df2.loc[df2[2]=='Direito da Saúde', 'Direito_Saúde']=1
df2.loc[df2[3]=='Direito da Saúde', 'Direito_Saúde']=1
df2.loc[df2[4]=='Direito da Saúde', 'Direito_Saúde']=1

df2.loc[df2[0]=='Direito do Consumidor', 'Direito_Consumidor']=1
df2.loc[df2[1]=='Direito do Consumidor', 'Direito_Consumidor']=1
df2.loc[df2[2]=='Direito do Consumidor', 'Direito_Consumidor']=1
df2.loc[df2[3]=='Direito do Consumidor', 'Direito_Consumidor']=1
df2.loc[df2[4]=='Direito do Consumidor', 'Direito_Consumidor']=1

df2.loc[df2[0]=='Direito Monetário', 'Direito_Monetário']=1
df2.loc[df2[1]=='Direito Monetário', 'Direito_Monetário']=1
df2.loc[df2[2]=='Direito Monetário', 'Direito_Monetário']=1
df2.loc[df2[3]=='Direito Monetário', 'Direito_Monetário']=1
df2.loc[df2[4]=='Direito Monetário', 'Direito_Monetário']=1

df2.loc[df2[0]=='Direito Comercial', 'Direito_Comercial']=1
df2.loc[df2[1]=='Direito Comercial', 'Direito_Comercial']=1
df2.loc[df2[2]=='Direito Comercial', 'Direito_Comercial']=1
df2.loc[df2[3]=='Direito Comercial', 'Direito_Comercial']=1
df2.loc[df2[4]=='Direito Comercial', 'Direito_Comercial']=1

df2.loc[df2[0]=='Direito Empresarial', 'Direito_Empresarial']=1
df2.loc[df2[1]=='Direito Empresarial', 'Direito_Empresarial']=1
df2.loc[df2[2]=='Direito Empresarial', 'Direito_Empresarial']=1
df2.loc[df2[3]=='Direito Empresarial', 'Direito_Empresarial']=1
df2.loc[df2[4]=='Direito Empresarial', 'Direito_Empresarial']=1

df2.loc[df2[0]=='Direito Agrário', 'Direito_Agrário']=1
df2.loc[df2[1]=='Direito Agrário', 'Direito_Agrário']=1
df2.loc[df2[2]=='Direito Agrário', 'Direito_Agrário']=1
df2.loc[df2[3]=='Direito Agrário', 'Direito_Agrário']=1
df2.loc[df2[4]=='Direito Agrário', 'Direito_Agrário']=1

df2.loc[df2[0]=='Direito Notarial e Registral', 'Direito_Notarial_Registral']=1
df2.loc[df2[1]=='Direito Notarial e Registral', 'Direito_Notarial_Registral']=1
df2.loc[df2[2]=='Direito Notarial e Registral', 'Direito_Notarial_Registral']=1
df2.loc[df2[3]=='Direito Notarial e Registral', 'Direito_Notarial_Registral']=1
df2.loc[df2[4]=='Direito Notarial e Registral', 'Direito_Notarial_Registral']=1

df2 = df2.drop([0,1,2,3,4], axis=1)
df2 = df2.fillna(0)
df2 = df2.astype(bool)
df2 = df2.reset_index(drop=False)
df2.rename(columns={'index':'Indice_Notícia'}, inplace=True)
df2

Indice_Notícia  Direito_Constitucional  Direito_Processual_Penal  \
0                   0                   False                     False   
1                   1                   False                     False   
2                   2                   False                     False   
3                   3                    True                     False   
4                   4                   False                     False   
...               ...                     ...                       ...   
10972           10972                   False                     False   
10973           10973                   False                      True   
10974           10974                   False                     False   
10975           10975                   False                     False   
10976           10976                    True                     False   

       Direito_Administrativo  Direito_Penal  Direito_Processual_Civil  \
0                        True          False                     False   
1                       False          False                     False   
2                        True          False                     False   
3                        True          False                     False   
4                       False          False                     False   
...                       ...            ...                       ...   
10972                    True          False                     False   
10973                   False          False                     False   
10974                   False           True                     False   
10975                    True          False                     False   
10976                    True          False                     False   

       Direito_Tributário  Direito_Previdenciário  Direito_Eleitoral  \
0                   False                   False              False   
1                    True                   False              False   
2                   False                   False              False   
3                   False                   False              False   
4                    True                   False              False   
...                   ...                     ...                ...   
10972               False                   False              False   
10973               False                   False              False   
10974               False                   False              False   
10975               False                   False              False   
10976               False                   False              False   

       Direito_Internacional  ...  Direito_Criança_Adolescente  \
0                      False  ...                        False   
1                      False  ...                        False   
2                      False  ...                        False   
3                      False  ...                        False   
4                      False  ...                        False   
...                      ...  ...                          ...   
10972                  False  ...                        False   
10973                  False  ...                        False   
10974                  False  ...                        False   
10975                  False  ...                        False   
10976                  False  ...                        False   

       Direito_Ambiental  Direito_Processual_Trabalho  Direito_Saúde  \
0                  False                        False          False   
1                  False                        False          False   
2                  False                        False          False   
3                  False                        False          False   
4                  False                        False          False   
...                  ...                          ...            ...   
10972              False                        False  

 > * Cada Ramo do Direito foi transformado em uma coluna categórica, possibilitando a realização de filtragem por tipo de Ramo do Direito.
 > * Criamos uma coluna "Índice de Notícia" de modo a atrelar cada notícia de julgamento a um número.

In [35]:
df1 = df['Ramo_Direito'].astype(str)
df3=df1.str.split(";", expand=True)
df3

columns=[0,1,2,3,4]
for i in columns:
    df3.loc[df3[i]=='Direito Constitucional', 'Direito_Constitucional']=1
    df3.loc[df3[i]=='Direito Processual Penal', 'Direito_Processual_Penal']=1
    df3.loc[df3[i]=='Direito Administrativo', 'Direito_Administrativo']=1
    df3.loc[df3[i]=='Direito Penal', 'Direito_Penal']=1
    df3.loc[df3[i]=='Direito Processual Civil', 'Direito_Processual_Civil']=1
    df3.loc[df3[i]=='Direito Tributário', 'Direito_Tributário']=1
    df3.loc[df3[i]=='Direito Previdenciário', 'Direito_Previdenciário']=1
    df3.loc[df3[i]=='Direito Eleitoral', 'Direito_Eleitoral']=1
    df3.loc[df3[i]=='Direito Internacional', 'Direito_Internacional']=1
    df3.loc[df3[i]=='Direito do Trabalho', 'Direito_Trabalho']=1
    df3.loc[df3[i]=='Direito Penal Militar', 'Direito_Penal_Militar']=1
    df3.loc[df3[i]=='Direito Processual Penal Militar', 'Direito_Processual_Penal_Militar']=1
    df3.loc[df3[i]=='Direito Civil', 'Direito_Civil']=1
    df3.loc[df3[i]=='Direito Financeiro', 'Direito_Financeiro']=1
    df3.loc[df3[i]=='Direito da Criança e do Adolescente', 'Direito_Criança_Adolescente']=1
    df3.loc[df3[i]=='Direito Ambiental', 'Direito_Ambiental']=1
    df3.loc[df3[i]=='Direito Processual do Trabalho', 'Direito_Processual_Trabalho']=1
    df3.loc[df3[i]=='Direito da Saúde', 'Direito_Saúde']=1
    df3.loc[df3[i]=='Direito do Consumidor', 'Direito_Consumidor']=1
    df3.loc[df3[i]=='Direito Monetário', 'Direito_Monetário']=1
    df3.loc[df3[i]=='Direito Comercial', 'Direito_Comercial']=1
    df3.loc[df3[i]=='Direito Empresarial', 'Direito_Empresarial']=1
    df3.loc[df3[i]=='Direito Agrário', 'Direito_Agrário']=1
    df3.loc[df3[i]=='Direito Notarial e Registral', 'Direito_Notarial_Registral']=1
    df3 = df3.drop([i], axis=1)
df3 = df3.fillna(0)
df3 = df3.astype(bool)
df3 = df3.reset_index(drop=False)
df3.rename(columns={'index':'Indice_Notícia'}, inplace=True)
df3

Indice_Notícia  Direito_Constitucional  Direito_Processual_Penal  \
0                   0                   False                     False   
1                   1                   False                     False   
2                   2                   False                     False   
3                   3                    True                     False   
4                   4                   False                     False   
...               ...                     ...                       ...   
10972           10972                   False                     False   
10973           10973                   False                      True   
10974           10974                   False                     False   
10975           10975                   False                     False   
10976           10976                    True                     False   

       Direito_Administrativo  Direito_Penal  Direito_Processual_Civil  \
0                        True          False                     False   
1                       False          False                     False   
2                        True          False                     False   
3                        True          False                     False   
4                       False          False                     False   
...                       ...            ...                       ...   
10972                    True          False                     False   
10973                   False          False                     False   
10974                   False           True                     False   
10975                    True          False                     False   
10976                    True          False                     False   

       Direito_Tributário  Direito_Previdenciário  Direito_Eleitoral  \
0                   False                   False              False   
1                    True                   False              False   
2                   False                   False              False   
3                   False                   False              False   
4                    True                   False              False   
...                   ...                     ...                ...   
10972               False                   False              False   
10973               False                   False              False   
10974               False                   False              False   
10975               False                   False              False   
10976               False                   False              False   

       Direito_Internacional  ...  Direito_Criança_Adolescente  \
0                      False  ...                        False   
1                      False  ...                        False   
2                      False  ...                        False   
3                      False  ...                        False   
4                      False  ...                        False   
...                      ...  ...                          ...   
10972                  False  ...                        False   
10973                  False  ...                        False   
10974                  False  ...                        False   
10975                  False  ...                        False   
10976                  False  ...                        False   

       Direito_Ambiental  Direito_Processual_Trabalho  Direito_Saúde  \
0                  False                        False          False   
1                  False                        False          False   
2                  False                        False          False   
3                  False                        False          False   
4                  False                        False          False   
...                  ...                          ...            ...   
10972              False                        False  

In [36]:
df3[df3['Direito_Constitucional']==True]

Indice_Notícia  Direito_Constitucional  Direito_Processual_Penal  \
3                   3                    True                     False   
5                   5                    True                     False   
6                   6                    True                     False   
7                   7                    True                     False   
8                   8                    True                     False   
...               ...                     ...                       ...   
10943           10943                    True                     False   
10944           10944                    True                     False   
10963           10963                    True                     False   
10965           10965                    True                     False   
10976           10976                    True                     False   

       Direito_Administrativo  Direito_Penal  Direito_Processual_Civil  \
3                        True          False                     False   
5                       False          False                     False   
6                       False          False                     False   
7                       False          False                     False   
8                       False          False                     False   
...                       ...            ...                       ...   
10943                   False          False                     False   
10944                   False          False                     False   
10963                   False          False                     False   
10965                   False          False                     False   
10976                    True          False                     False   

       Direito_Tributário  Direito_Previdenciário  Direito_Eleitoral  \
3                   False                   False              False   
5                   False                   False              False   
6                   False                   False               True   
7                   False                   False               True   
8                   False                   False               True   
...                   ...                     ...                ...   
10943               False                   False              False   
10944               False                   False              False   
10963               False                   False              False   
10965               False                   False              False   
10976               False                   False              False   

       Direito_Internacional  ...  Direito_Criança_Adolescente  \
3                      False  ...                        False   
5                      False  ...                        False   
6                      False  ...                        False   
7                      False  ...                        False   
8                      False  ...                        False   
...                      ...  ...                          ...   
10943                  False  ...                        False   
10944                  False  ...                        False   
10963                  False  ...                        False   
10965                  False  ...                        False   
10976                  False  ...                        False   

       Direito_Ambiental  Direito_Processual_Trabalho  Direito_Saúde  \
3                  False                        False          False   
5                  False                        False          False   
6                  False                        False          False   
7                  False                        False          False   
8                  False                        False          False   
...                  ...                          ...            ...   
10943              False                        False  

In [37]:
df3

Indice_Notícia  Direito_Constitucional  Direito_Processual_Penal  \
0                   0                   False                     False   
1                   1                   False                     False   
2                   2                   False                     False   
3                   3                    True                     False   
4                   4                   False                     False   
...               ...                     ...                       ...   
10972           10972                   False                     False   
10973           10973                   False                      True   
10974           10974                   False                     False   
10975           10975                   False                     False   
10976           10976                    True                     False   

       Direito_Administrativo  Direito_Penal  Direito_Processual_Civil  \
0                        True          False                     False   
1                       False          False                     False   
2                        True          False                     False   
3                        True          False                     False   
4                       False          False                     False   
...                       ...            ...                       ...   
10972                    True          False                     False   
10973                   False          False                     False   
10974                   False           True                     False   
10975                    True          False                     False   
10976                    True          False                     False   

       Direito_Tributário  Direito_Previdenciário  Direito_Eleitoral  \
0                   False                   False              False   
1                    True                   False              False   
2                   False                   False              False   
3                   False                   False              False   
4                    True                   False              False   
...                   ...                     ...                ...   
10972               False                   False              False   
10973               False                   False              False   
10974               False                   False              False   
10975               False                   False              False   
10976               False                   False              False   

       Direito_Internacional  ...  Direito_Criança_Adolescente  \
0                      False  ...                        False   
1                      False  ...                        False   
2                      False  ...                        False   
3                      False  ...                        False   
4                      False  ...                        False   
...                      ...  ...                          ...   
10972                  False  ...                        False   
10973                  False  ...                        False   
10974                  False  ...                        False   
10975                  False  ...                        False   
10976                  False  ...                        False   

       Direito_Ambiental  Direito_Processual_Trabalho  Direito_Saúde  \
0                  False                        False          False   
1                  False                        False          False   
2                  False                        False          False   
3                  False                        False          False   
4                  False                        False          False   
...                  ...                          ...            ...   
10972              False                        False  

In [38]:
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10977 entries, 0 to 10976
Data columns (total 25 columns):
 #   Column                            Non-Null Count  Dtype
---  ------                            --------------  -----
 0   Indice_Notícia                    10977 non-null  int64
 1   Direito_Constitucional            10977 non-null  bool 
 2   Direito_Processual_Penal          10977 non-null  bool 
 3   Direito_Administrativo            10977 non-null  bool 
 4   Direito_Penal                     10977 non-null  bool 
 5   Direito_Processual_Civil          10977 non-null  bool 
 6   Direito_Tributário                10977 non-null  bool 
 7   Direito_Previdenciário            10977 non-null  bool 
 8   Direito_Eleitoral                 10977 non-null  bool 
 9   Direito_Internacional             10977 non-null  bool 
 10  Direito_Trabalho                  10977 non-null  bool 
 11  Direito_Penal_Militar             10977 non-null  bool 
 12  Direito_Processual_Penal_Militar

In [39]:
df4=pd.concat([df, df3], axis=1)

In [40]:
df5 = df4[['Informativo','Indice_Notícia','Ano','Mês', 'Dia',
           'Direito_Constitucional', 'Direito_Processual_Penal', 'Direito_Administrativo','Direito_Penal',
           'Direito_Processual_Civil', 'Direito_Tributário', 'Direito_Previdenciário', 'Direito_Eleitoral',
           'Direito_Internacional', 'Direito_Trabalho', 'Direito_Penal_Militar', 'Direito_Processual_Penal_Militar', 
           'Direito_Civil', 'Direito_Financeiro', 'Direito_Criança_Adolescente', 'Direito_Ambiental',
           'Direito_Processual_Trabalho', 'Direito_Saúde', 'Direito_Consumidor', 'Direito_Monetário',
           'Direito_Comercial', 'Direito_Empresarial', 'Direito_Agrário', 'Direito_Notarial_Registral'
          ]]
df5

Informativo  Indice_Notícia   Ano Mês Dia  Direito_Constitucional  \
0             1127               0  2024  03  08                   False   
1             1127               1  2024  03  08                   False   
2             1127               2  2024  03  08                   False   
3             1127               3  2024  02  20                    True   
4             1127               4  2024  03  08                   False   
...            ...             ...   ...  ..  ..                     ...   
10972            1           10972  1995  08  16                   False   
10973            1           10973  1995  08  08                   False   
10974            1           10974  1995  08  08                   False   
10975            1           10975  1995  08  08                   False   
10976            1           10976  1995  08  09                    True   

       Direito_Processual_Penal  Direito_Administrativo  Direito_Penal  \
0                         False                    True          False   
1                         False                   False          False   
2                         False                    True          False   
3                         False                    True          False   
4                         False                   False          False   
...                         ...                     ...            ...   
10972                     False                    True          False   
10973                      True                   False          False   
10974                     False                   False           True   
10975                     False                    True          False   
10976                     False                    True          False   

       Direito_Processual_Civil  ...  Direito_Criança_Adolescente  \
0                         False  ...                        False   
1                         False  ...                        False   
2                         False  ...                        False   
3                         False  ...                        False   
4                         False  ...                        False   
...                         ...  ...                          ...   
10972                     False  ...                        False   
10973                     False  ...                        False   
10974                     False  ...                        False   
10975                     False  ...                        False   
10976                     False  ...                        False   

       Direito_Ambiental  Direito_Processual_Trabalho  Direito_Saúde  \
0                  False                        False          False   
1                  False                        False          False   
2                  False                        False          False   
3                  False                        False          False   
4                  False                        False          False   
...                  ...                          ...            ...   
10972              False                        False          False   
10973              False                        False          False   
10974              False                        False          False   
10975              False                        False          False   
10976              False                        False          False   

       Direito_Consumidor  Direito_Monetário  Direito_Comercial  \
0                   False              False              False   
1                   False              False              False   
2                   False              False              False   
3                   False              False              False   
4                   False              False              False   
...                   ...                ...                ...   
10972               False              Fal

In [41]:
df5 = df5[df5['Direito_Constitucional']==True]
df5 = df5[df5['Direito_Processual_Penal']==True]
df5



Informativo  Indice_Notícia   Ano Mês Dia  Direito_Constitucional  \
9             1126               9  2024  03  01                    True   
21            1123              21  2024  02  06                    True   
63            1117              63  2023  11  20                    True   
65            1117              65  2023  11  20                    True   
103           1110             103  2023  09  29                    True   
...            ...             ...   ...  ..  ..                     ...   
10833           14           10833  1995  11  14                    True   
10887            9           10887  1995  10  11                    True   
10900            8           10900  1995  10  03                    True   
10909            7           10909  1995  09  26                    True   
10920            7           10920  1995  09  26                    True   

       Direito_Processual_Penal  Direito_Administrativo  Direito_Penal  \
9                          True                   False           True   
21                         True                   False          False   
63                         True                   False           True   
65                         True                   False          False   
103                        True                   False          False   
...                         ...                     ...            ...   
10833                      True                   False          False   
10887                      True                   False           True   
10900                      True                   False          False   
10909                      True                   False          False   
10920                      True                   False          False   

       Direito_Processual_Civil  ...  Direito_Criança_Adolescente  \
9                         False  ...                        False   
21                        False  ...                        False   
63                        False  ...                        False   
65                        False  ...                        False   
103                       False  ...                        False   
...                         ...  ...                          ...   
10833                     False  ...                        False   
10887                     False  ...                        False   
10900                     False  ...                        False   
10909                     False  ...                        False   
10920                     False  ...                        False   

       Direito_Ambiental  Direito_Processual_Trabalho  Direito_Saúde  \
9                  False                        False          False   
21                 False                        False          False   
63                 False                        False          False   
65                 False                        False          False   
103                False                        False          False   
...                  ...                          ...            ...   
10833              False                        False          False   
10887              False                        False          False   
10900              False                        False          False   
10909              False                        False          False   
10920              False                        False          False   

       Direito_Consumidor  Direito_Monetário  Direito_Comercial  \
9                   False              False              False   
21                  False              False              False   
63                  False              False              False   
65                  False              False              False   
103                 False              False              False   
...                   ...                ...                ...   
10833               False              Fal

> * Observe-se que o df5 é um DataFrame que apresenta número do Informativo, o ìndice_Notícia e os Ramos do Direito abordados.

In [47]:
df6=df4[['Indice_Notícia', 'Informativo', 'Classe_Processo',
        'Número_Processo','Incidente_Julgamento','UF','Observação',
        'Data_Julgamento', 'Relator', 'Redator_Acórdão', 'Órgão Julgador', 'Tipo_Julgamento', 
        'Situação_Julgamento','Título', 'Tese_Julgado', 'Resumo', 
        'Notícia', 'Ramo_Direito', 'Matéria', 'Repercussão_Geral','Tema_RG', 'Legislação', 
         'ODS_ONU_2030', 'Covid-19', 'Notícia_Completa' 
                 ]]
df6.head(2)

Indice_Notícia  Informativo Classe_Processo  Número_Processo  \
0               0         1127             ADI             7218   
1               1         1127             ADI             7239   

  Incidente_Julgamento  UF Observação      Data_Julgamento  \
0                  nan  PB        nan  2024-03-08 03:00:00   
1                  nan  DF        nan  2024-03-08 03:00:00   

                     Relator Redator_Acórdão  ...  \
0          MIN. DIAS TOFFOLI             nan  ...   
1  MIN. LUÍS ROBERTO BARROSO             nan  ...   

                                              Resumo  \
0  São inconstitucionais — por ofensa ao princípi...   
1  É constitucional a exclusão do regime de isenç...   

                                             Notícia            Ramo_Direito  \
0  Conforme a jurisprudência desta Corte, o exerc...  Direito Administrativo   
1  As alterações promovidas pelo dispositivo impu...      Direito Tributário   

                                             Matéria Repercussão_Geral  \
0  Organização Político-Administrativa; Administr...               Não   
1  Zona Franca de Manaus; isenção fiscal; áreas d...               Não   

  Tema_RG                                         Legislação  \
0       0  CF/1988: arts. 132, caput e 207.\nADCT: art. 6...   
1       0  Decreto-Lei nº 288/1967: art. 37\nLei nº 14.18...   

                                 ODS_ONU_2030 Covid-19  \
0  8 Trabalho Decente e Crescimento Econômico      Não   
1                                         nan      Não   

                                    Notícia_Completa  
0  São inconstitucionais — por ofensa ao princípi...  
1  “É constitucional o dispositivo de lei federal...  

[2 rows x 25 columns]

> * Observe-se que o df6 é o DataFrame que apresenta informações com conteúdo da notícia vinculado em cada julgamento publicado no Informativo do STF.
> * Com base no df5 é possível filtrar os julgados por ramo do Direito e atravês do Indice_Notícia apresentada após o filtro, é possível fazer a busca no df6, aonde consta as notícias completas do julgamento.

Abaixo, apresenta-se o codigo para se fazer a busca da notícia completa, de acordo com o Indice da Notícia que equivale aos index do df6

In [48]:
df1 = df6.loc[0:1,'Notícia_Completa']
print(df1.values[1])
df1

“É constitucional o dispositivo de lei federal que tão somente explicita a extensão dos benefícios fiscais concedidos à Zona Franca de Manaus pelo Decreto-Lei nº 288/1967, em sua redação original.” 

É constitucional a exclusão do regime de isenção fiscal da Zona Franca de Manaus das exportações ou reexportações, importações e operações com petróleo, lubrificantes e combustíveis líquidos e gasosos derivados de petróleo. 

As alterações promovidas pelo dispositivo impugnado apenas explicitam as exceções ao tratamento fiscal favorecido, originalmente disposto no art. 37 do Decreto-Lei nº 288/1967 (1) e cujo objetivo é preservar o desenvolvimento regional, por meio da indução de investimentos na região. Elas não alteraram o conjunto de benefícios e incentivos fiscais da Zona Franca de Manaus e, consequentemente, a proteção constitucional a ela conferida (2). 

Nesse contexto, a exclusão do regime fiscal das atividades que envolvam petróleo e seus derivados visa neutralizar a assimetria tr

0    São inconstitucionais — por ofensa ao princípi...
1    “É constitucional o dispositivo de lei federal...
Name: Notícia_Completa, dtype: object

In [49]:
df2 = df6.loc[0:0,'Notícia_Completa']
print(df2.values[0])

São inconstitucionais — por ofensa ao princípio da unicidade orgânica da advocacia pública estadual (CF/1988, art. 132, caput) — normas locais que preveem cargos e carreiras de advogado ou de procurador para viabilizar a criação ou a manutenção de órgãos de assessoramento jurídico no âmbito de autarquias e fundações estaduais. Esse entendimento não se aplica, dentre outros casos, na hipótese de instituição de procuradorias em universidades estaduais e de manutenção dos órgãos de consultoria jurídica já existentes na data da promulgação da Constituição Federal de 1988 (ADCT, art. 69).

Conforme a jurisprudência desta Corte, o exercício das atividades de representação judicial e de consultoria jurídica no âmbito dos estados e do Distrito Federal é de competência exclusiva dos Procuradores do Estado (1). 
Por outro lado, este Tribunal reconhece, de modo restritivo, algumas exceções à mencionada regra: (i) instituição de procuradorias em universidades estaduais em razão do princípio da aut

Após esta análise exploratória do banco de dados de Informativos do STF já é possível apresentar um projeto Streamlit de filtragem dos julgados por meio do ramo do direito abordado por cada um dos julgados, facilitando ao estudante de Direito ou ao Profissional da área filtrar os julgados de todos os Informativos já publicados, de acordo com o seu interesse e, de forma ágil, obter o contéudo da Notícia.